build/changelog/ChangelogBuilder.php

```diff
@@ -6,62 +6,72 @@
  */
 class ChangelogBuilder
 {
+    /** @var array */
+    private $params = [];
 
+    /** @var boolean */
     private $newServiceFlag = false;
-    private $baseDir;
-    private $releaseNotesOutputDir;
-    private $verboseFlag;
 
-    public function __construct($baseDir, $releaseNotesOutputDir, $verboseFlag)
+    /**
+     *  The constructor requires following configure parameters:
+     * - base_dir: (String) Path to the base directory where the `.changes` folder is located. Default is empty string.
+     * - release_notes_output_dir: (String) Path to `.changes` folder where new release json will be put and content of
+     *                              `nextrelease` folder will be deleted
+     * - verbose_flag: (Boolean) Flag to enable(true)/disable(false) verbose mode
+     */
+    public function __construct(array $params)
     {
-        $this->baseDir = $baseDir;
-        $this->releaseNotesOutputDir = $releaseNotesOutputDir;
-        $this->verboseFlag = $verboseFlag;
+        $this->params['base_dir'] = isset($params['base_dir']) ? $params['base_dir'] : '';
```

Comment:
I think it's much clearer to bind values pulled from the parameters array to instance properties instead of putting them in an internal parameters array. Is there any advantage to using an array instead of instance properties?